registry/coder/modules/agentapi/main.tf

```diff
@@ -126,6 +126,27 @@ variable "agentapi_port" {
   default     = 3284
 }
 
+locals {
+  # agentapi_subdomain_false_min_version_expr matches a semantic version >= v0.3.1.
+  agentapi_subdomain_false_min_version_expr = "^v(0\\.(3\\.[1-9]+|[4-9]+\\.\\d+)|[1-9]\\d*\\.\\d+\\.\\d+)$"
```

Comment:
Why this? Wouldn't the module simply fail if someone tries to set it on an older version of the module? As the input variable simply does not exist.

Reply:
I think the input would be invalid for older module versions, So this validation would never be hit.

Reply:
NVM. I was confusing the agentAPI version with the AgnetAPI module versions. Please consider this resolved.

Reply:
Ah I understand the confusion now :-) Without this, we could end up passing an invalid CLI argument to the `agentapi` binary. However, @hugodutka has added the ability to [configure agentapi via env](https://github.com/coder/agentapi/pull/44) which will reduce the impact here. Nonetheless I feel this will help prevent confusion.